File: schemaorggo/gameserver.go
package schemaorggo

import "encoding/json"

// GameServer see : https://schema.org/GameServer
type GameServer struct {
	Intangible

	typeContext

	// Game see : https://schema.org/game
	// Video game which is played on this server. Inverse property: gameServer (see: https://schema.org/gameServer).
	Game *VideoGame `json:"game,omitempty"` // types : VideoGame

	// PlayersOnline see : https://schema.org/playersOnline
	// Number of players on the server.
	PlayersOnline float64 `json:"playersOnline,omitempty"` // types : Integer

	// ServerStatus see : https://schema.org/serverStatus
	// Status of a game server.
	ServerStatus *GameServerStatus `json:"serverStatus,omitempty"` // types : GameServerStatus

}

func (v GameServer) MarshalJSONWithTypeContext() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "GameServer"

	return json.Marshal(v)
}

func (v *GameServer) MarshalJSON() ([]byte, error) {
	if v == nil {
		return []byte("null"), nil
	}

	v.C = "http://schema.org"
	v.T = "GameServer"

	return json.Marshal(*v)
}

File: schemaorgphp/classes/game_server.php
<?php

class GameServer extends Intangible implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'GameServer';
	
	/**
	 * Video game which is played on this server. Inverse property: gameServer (see: https://schema.org/gameServer).
	 * see : https://schema.org/game
	 * @var \VideoGame|\VideoGame[]
	 */
	public var $game;
	
	/**
	 * Number of players on the server.
	 * see : https://schema.org/playersOnline
	 * @var integer|integer[]
	 */
	public var $players_online;
	
	/**
	 * Status of a game server.
	 * see : https://schema.org/serverStatus
	 * @var \GameServerStatus|\GameServerStatus[]
	 */
	public var $server_status;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'GameServer'
		);
		
		$serialized = so_json_serialize( $this->game );
		if ( ! empty( $serialized ) ) {
			$out['game'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->players_online );
		if ( ! empty( $serialized ) ) {
			$out['playersOnline'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->server_status );
		if ( ! empty( $serialized ) ) {
			$out['serverStatus'] = $serialized;
		}
		
		return $out;
	}
}
